pkg/scheduler/schedule_one.go

```diff
@@ -88,13 +86,37 @@ func (sched *Scheduler) scheduleOne(ctx context.Context) {
 	start := time.Now()
 	state := framework.NewCycleState()
 	state.SetRecordPluginMetrics(rand.Intn(100) < pluginMetricsSamplePercent)
+
 	// Initialize an empty podsToActivate struct, which will be filled up by plugins or stay empty.
 	podsToActivate := framework.NewPodsToActivate()
 	state.Write(framework.PodsToActivateKey, podsToActivate)
 
 	schedulingCycleCtx, cancel := context.WithCancel(ctx)
 	defer cancel()
-	scheduleResult, err := sched.SchedulePod(schedulingCycleCtx, fwk, state, pod)
+	scheduleResult, assumedPodInfo, err := sched.schedulingCycle(schedulingCycleCtx, state, fwk, podInfo, podsToActivate, start)
+	if err != nil {
+		// We don't need to do anything with a returned error here because it is properly handled in schedulingCycle().
```

Comment:
Actually, instead of calling `FailureHandler` in every single place, we could call it here. Kind of similar idea to https://github.com/kubernetes/kubernetes/issues/103853#issuecomment-889370324

I originally proposed we used defer, but it might just make sense to do it here. Sure, we need to use `schedulingCycleCtx` twice, but that's not a lot to manage.

Reply:
👍 do that in 08cce7a.

Reply:
the comment is outdated